examples/bq-email-exports/README.md

```diff
@@ -1,38 +1,34 @@
-**Cloud Storage:** A [Cloud Storage](https://cloud.google.com/storage/) bucket stores the CSV file. The Cloud Storage API generates a signed URL for the CSV.  
+**Cloud Function:** A [Cloud Function](https://cloud.google.com/functions) subscribes to the Pub/Sub topic and runs the code calling the BigQuery and Cloud Storage APIs. 
+**BigQuery:** The [BigQuery API](https://cloud.google.com/bigquery/docs/reference/rest) generates the query results, stores them in a table, and then exports the results as a JSON file into Cloud Storage.  
+**Cloud Storage:** A [Cloud Storage](https://cloud.google.com/storage/) bucket stores the JSON file. The Cloud Storage API generates a signed URL for the JSON file.  
 **SendGrid API** The [SendGrid API](https://sendgrid.com/) is a web based API that sends the signed URL as an email to users.
 
 To implement this solution, follow the steps below:
 
 ## Set Up
-1. Create a BigQuery Dataset and set a default expiration time of the table to prevent storing outdated data.
-```bash
-bq --location=US mk -d --default_table_expiration 3600 bq_exports
-```
-2. Create a Google Cloud Storage bucket to host the exported CSV files from BigQuery.
-```bash   
-gsutil mb gs://bq_email_exports/
-```
-3. Generate a SendGrid API key by creating a free tier [SendGrid account](https://signup.sendgrid.com/).
+1. Generate a SendGrid API key by creating a free tier [SendGrid account](https://signup.sendgrid.com/).
+
+2. Save the query you intend to run in a Cloud Storage bucket as a text file. Specify the bucket and file name in `terraform/payload.txt` so the Cloud Function what query to run.
+
+3. Set the remaining variables in `terraform/payload.txt` to the appropriate values. This payload will be sent from Cloud Scheduler to the function. This way, the same Cloud Function can be used for different queries and emails by simply updating the payload. Keep in mind that the `table_name` in `terraform/payload.txt` will automatically have a timestamp appended to the end from the Cloud Function code.
 
 ## Deploying the pipeline
 
-To deploy the pipeline, first update the variables in `deploy.sh` and `main.py`. Then, run:
+To deploy the pipeline, run:
 ```bash
-./deploy.sh
+cd terraform
+terraform init
+terraform apply
 ```
-This file creates a service account with the BigQuery Admin, Storage Object Admin, and Service Account Token Creator roles. These roles are needed to run the query, generate the signed URL, and generate the signing credentials to authenticate the `generate_signed_url()` function.
-
-It also creates the Pub/Sub topic to trigger the Cloud Function, deploys the Cloud Function using `main.py`, and creates a Cloud Scheduler job to run the function on a regular interval. 
+The Cloud Function that will be deployed uses the compressed zip file (`terraform/main.zip`) containing `main.py` and `requirements.txt` as the source code.
 
```

Comment:
please add a section of caveats / considerations:
- This won't handle queries that run longer than ~9 mins
- This won't handle large results sizes in a single file
- Signed URLs are a data exfiltration risk
- everyone should be a gsuite customer and use the sheets solution as it's much better for everyone (kidding don't add this)

Also add alternative approaches if use case doesn't meet above constraints like doing this with a simple airflow DAG.